### DISCUSSION TITLE
Migrate from v0.3.4 to v0.4.0 on Lanbon L8

### DISCUSSION THREAD

**Discussion by Polomans** [HASwitchPlate/openHASP#116] [Q&A] [answered]
Can you please tell me if it is possible to update by air? It's just that the list of files contains lanbon_l8_**ota**_v0.4.0.bin

And if not, then where is the new partitions.bin or is it already in lanbon_l8_**full**_8MB_v0.4.0.bin ?

**Comment by fvanroie** [comment, marked as answer]
There are 2 binary files available for each device:
- `ota` only updates the application partition on the device
- `full` contains *all* partitions at once for easy flashing

The v0.4.0 ota binary will not fit on the v0.3.4 application partition, so you need to flash the full binary first. The full binary is an all-in-one containing the bootloader, application and appropriate partition schema for that device. There is no more need to flash those separately.


## Install the firmware
### Flash ESP32

When flashing the ESP32 for the first time, you need to flash it over serial using the full firmware binary file:

```bash
esptool.py --port COM1 erase_flash
esptool.py --port COM1 --baud 921600 write_flash 0x0 lanbon_l8_full_8MB_v0.4.0.bin
```
Change COM1 to the correct port on your computer.
